modules/packages/LinearAlgebra.chpl

```diff
@@ -0,0 +1,709 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Linear Algebra Module
+
+A high-level interface to linear algebra operations and procedures.
+
+.. note::
+  This is an early prototype package module. As a result, interfaces may change
+  over the next release.
+
+Compiling with Linear Algebra
+-----------------------------
+
+The linear algebra module uses both `BLAS` and `LAPACK` for
+many routines. In order to compile a Chapel program with this module, be sure
+to have BLAS and LAPACK implementations available on your system. See the BLAS
+and LAPACK documentation for further details.
+
+
+Linear Algebra API
+------------------
+
+**Matrix and Vector representation**
+
+Matrices and vectors are represented as Chapel arrays, which means the
+convenience constructors provided in this module are not necessary to use this
+module's functions. This also means that matrices (2D arrays) and vectors (1D
+arrays) will work with any other Chapel library works with arrays.
+
+.. note::
+
+  This documentation uses the terms `matrix` to refer to `2D arrays`, and
+  `vector` to refer to `1D arrays`.
+
+**Indexing**
+
+All functions that return new arrays will return arrays over 0-based indices,
+unless otherwise specified.
+
+**Matrix multiplication**
+
+``dot()`` is the general matrix multiplication function provided in this module.
+This function supports any combination of scalars, vectors (1D arrays), and
+matrices (2D arrays). See the :proc:`dot` documentation for more information.
+
+If you find this convenient notation confusing, you can opt to calling the
+underlying functions: :proc:`matMult` and `inner`.
+
+The :proc:`dot` function, along with others may be given a matrix-specific
+operator in future releases.
+
+.. comment:
+  It is also less verbose than matrixMultiply, matMult, or mMult.
```

Comment:
I was on the border of uncommenting this to describe why `dot` was chosen as the general matrix multiplication function as opposed to other names... but I also don't want the documentation to sound defensive.